FLT/NumberField/AdeleRing.lean

```diff
@@ -30,26 +34,47 @@ section BaseChange
 
 namespace NumberField.AdeleRing
 
+open IsDedekindDomain
+
 variable (K L : Type*) [Field K] [NumberField K] [Field L] [NumberField L] [Algebra K L]
 
 scoped notation:100 "𝔸" K => AdeleRing (𝓞 K) K
 
 noncomputable instance : Algebra K (𝔸 L) :=
-  Algebra.compHom _ (algebraMap K L)
+  inferInstanceAs (Algebra K (InfiniteAdeleRing L × FiniteAdeleRing (𝓞 L) L))
 
 instance : IsScalarTower K L (𝔸 L) :=
-  IsScalarTower.of_algebraMap_eq' rfl
+  IsScalarTower.of_algebraMap_eq fun _ ↦ rfl
 
 /-- The canonical map from the adeles of K to the adeles of L -/
 noncomputable def baseChange :
     (𝔸 K) →A[K] 𝔸 L :=
-  sorry -- product of finite and infinite adele maps
+  let finite : FiniteAdeleRing (𝓞 K) K →A[K] FiniteAdeleRing (𝓞 L) L := {
```

Comment:
Do you think that really this should be a continuous semialgebra hom over K -> L?

Reply:
For the purposes of defining the base change isomorphism for ideals, this would be fine as a normal semialgebra homomorphism (and a lemma that it's continuous if we want that).